src/React.AspNet/HttpContextLifetimeProvider.cs

```diff
@@ -44,27 +44,13 @@ public HttpContextLifetimeProvider(IServiceProvider appServiceProvider)
 		private readonly string _keyName = PREFIX + Guid.NewGuid();
 
 		/// <summary>
-		/// Gets the <see cref="HttpContext" /> of the current request.
-		/// </summary>
-		private HttpContext HttpContext => 
-			_appServiceProvider.GetRequiredService<IHttpContextAccessor>().HttpContext;
-
-		/// <summary>
 		/// Gets the current per-request registrations for the current request.
 		/// </summary>
 		private PerRequestRegistrations Registrations
 		{
 			get
 			{
-				var requestServices = HttpContext.RequestServices;
-				if (requestServices == null)
-				{
-					throw new ReactNotInitialisedException(
-						"ASP.NET request services have not been initialised correctly. Please " +
-						"ensure you are calling app.UseRequestServices() before app.UseReact()."
-					);
-				}
-				var registrations = requestServices.GetService<PerRequestRegistrations>();
+				var registrations = _appServiceProvider.GetService<PerRequestRegistrations>();
```

Comment:
This change was incorrect, it wasn't handling per-request singletons properly and was instead just using a single instance for the entire app. I fixed it in a4992da25bd4880aab3b2fabe739e6a176dbde6f :+1: